### PR TITLE
[last-element] Added definitions

### PR DESCRIPTION
**Pull request by ChocolateLoverRaj**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/last-element/last-element-tests.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import last from 'last-element';
```
Comment (peterblazejewicz) [inline types/last-element/last-element-tests.ts:1]:
```suggestion
import last = require('last-element');
```

File: types/last-element/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/last-element/tsconfig.json:16]:
```suggestion
```
it's for consuming code, not here on DT (I'v echanged the import already)

File: types/last-element/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for last-element 1.0
+// Project: https://github.com/Prosen-Ghosh/last-element#readme
+// Definitions by: Rajas Paranjpe <https://github.com/ChocolateLoverRaj>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function last <T>(arr: readonly T[]): T;
```
Comment (peterblazejewicz) [inline types/last-element/index.d.ts:6]:
```suggestion
export as namespace last;

declare function last <T>(arr: readonly T[]): T;
```
It also works in the browser via script inclusion

**Comment by peterblazejewicz** [comment]
@ChocolateLoverRaj when accepted, can you open another PR that removes DT package, wihtout adding it to the 'no longer reuqired' list as on the README?
the package you're adding types for is readonly, and that's unlikely it will be ever updated.
So once DT publishes types, let's remove sources, so there is no impact on maintaining (there are thousands of DT here), thx!

File: types/last-element/index.d.ts (hunk 3 above)
Comment (ChocolateLoverRaj) [reply types/last-element/index.d.ts:6]: @peterblazejewicz 
So would doing it this way make `last` a global function?

**Comment by ChocolateLoverRaj** [comment]
@peterblazejewicz After this pull request is merged, should I delete the `last-element` folder and make another PR?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ChocolateLoverRaj thx!
Only after types are published to NPM, there will be an entry here, just reference me back, so we can continue, thx!

**Comment by ChocolateLoverRaj** [comment]
> LGTM!
> @ChocolateLoverRaj thx!
> Only after types are published to NPM, there will be an entry here, just reference me back, so we can continue, thx!

@peterblazejewicz The types have been published. What are the next steps?

**Pull request merged**